airflow/api_fastapi/core_api/datamodels/monitor.py

```diff
@@ -49,4 +51,8 @@ class HealthInfoSchema(BaseModel):
     metadatabase: BaseInfoSchema
     scheduler: SchedulerInfoSchema
     triggerer: TriggererInfoSchema
-    dag_processor: DagProcessorInfoSchema
+    dag_processor: DagProcessorInfoSchema | None = None
+
+    @model_serializer()
+    def serialize(self):
+        return {k: v for k, v in self if not (k == "dag_processor" and v is None)}
```

Comment:
@pierrejeambrun , can we use this(model_serializer) to implement fields to return in list dag_runs?

Reply:
If we use `response_model_exclude_unset=True`, we won't need it anymore.

Also just a [field_serializer](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.field_serializer) on the `dag_processor` is most likely enough.

Reply:
Sorry. I meant, we did not include fields query param in list dag runs as we did not want to have model with dynamic fields.